src/lib.rs

```diff
@@ -1,18 +1,10 @@
 use std::{fmt, hash, ops::Deref, sync::Arc};
 
-/// A `SmolStr` is a string type that has the following properties
+/// A `SmolStr` is a string type that has the following properties:
 ///
-///  * `size_of::<SmolStr>() == size_of::<String>()`
-///  * Strings up to 22 bytes long do not use heap allocations
-///  * Runs of `\n` and space symbols (typical whitespace pattern of indentation
-///    in programming laguages) do not use heap allocations
-///  * `Clone` is `O(1)`
-///
-/// Unlike `String`, however, `SmolStr` is immutable. The primary use-case for
-/// `SmolStr` is a good enough default storage for tokens of typical programming
-/// languages. A specialized interner might be a better solution for some use-cases.
-///
-/// Intenrally, `SmolStr` is roughly an `enum { Heap<Arc<str>>, Inline([u8; 22]) }`.
+/// * `size_of::<SmolStr>() == size_of::<String>()`
+/// * Strings of up to 22 bytes are stack-allocated (see the readme for details)
```

Comment:
I think it's better to copy-paste the info from readme over here: this is a doc string, so it is available when you browsing the docs on docs.rs or locally (`cargo doc --open`), and Readme is not necessary packaged up with the docs.